internal/runtimehandlerhooks/runtime_handler_hooks_linux.go

```diff
@@ -10,32 +10,52 @@ import (
 	libconfig "github.com/cri-o/cri-o/pkg/config"
 )
 
-// GetRuntimeHandlerHooks returns RuntimeHandlerHooks implementation by the runtime handler name.
-func GetRuntimeHandlerHooks(ctx context.Context, config *libconfig.Config, handler string, annotations map[string]string) (RuntimeHandlerHooks, error) {
+// NewMap creates a new Map of runtime names to runtime hooks from Crio's configuration.
+func NewMap(ctx context.Context, config *libconfig.Config) Map {
 	ctx, span := log.StartSpan(ctx)
 	defer span.End()
 
-	if strings.Contains(handler, HighPerformance) {
-		log.Warnf(ctx, "The usage of the handler %q without adding high-performance feature annotations under allowed_annotations will be deprecated under 1.21", HighPerformance)
+	var hphInstance *HighPerformanceHooks
 
-		return &HighPerformanceHooks{irqBalanceConfigFile: config.IrqBalanceConfigFile, cpusetLock: sync.Mutex{}, sharedCPUs: config.SharedCPUSet}, nil
-	}
+	rhh := make(Map)
 
-	if highPerformanceAnnotationsSpecified(annotations) {
-		log.Warnf(ctx, "The usage of the handler %q without adding high-performance feature annotations under allowed_annotations will be deprecated under 1.21", HighPerformance)
```

Comment:
Because of the change, the check is no longer done.
maybe we can check in each [Pre|Post][Start|Stop] function?

Reply:
I also changed when the check is printed: which is only when the runtime name contains "high-performance" and when the annotation is not specified. Before, the Warning was printed always when the runtime was named ".*high-performance.*", even when the annotations were present (even more log spamming IMO)

Reply:
> The check is still done, but it's done when crio starts and prints this exactly once per handler that's using the high-performance feature.

The code I highlighted checks the sandbox's annotation, not allowed_annotation. The sandbox's annotation can't be checked at cri-o startup time.

However, the older code, only checks the allowed_annotation https://github.com/cri-o/cri-o/commit/106e662ad70fbdc8c03d1ea02e5dc885fbfbaec7#diff-d9b3c6e0b55a2c65987888288a47e86232d3edea55062138ec2a7fef6128050d .
and considering the fact that the warning didn't work properly, we may be able to remove this check.

@haircommander do you know if we should check the sandbox annotation?

Reply:
be it as it may, though ... the `func (m Map) Get(name string) RuntimeHandlerHooks` method replaces `GetRuntimeHandlerHooks` where it was used before. I could just add the `annotations` as a parameter to `func (m Map) Get(name string) RuntimeHandlerHooks` and move the check for the sb annotation in there, that way it's in the same location as before and it's for the actual sandbox annotation.
Then, either return the value in `m[name]` if a highperformance annotation is present on the sandbox, or nil otherwise --> should then behave as before. 
I could still keep the `Warnf ` in its current location (or move it back to Get, I don't care). The rest of the code would remain exactly the same.

The question is if that's necessary

Reply:
ok, I added a unit test that should capture the cases that were discussed.

From my understanding, the following cases must be covered for Linux:
- runtime name contains "high-performance" OR sandbox annotation has high performance --> `highPerformanceHooks`
- cpuLoadBalancingAllowed anywhere --> DefaultCPULoadBalanceHooks
- nil otherwise

On other architectures: always DefaultCPULoadBalanceHooks

So having a precreated `map[string]RuntimeHandlerHooks` doesn't make any sense; therefore I went back to something similar to an earlier iteration without this map.